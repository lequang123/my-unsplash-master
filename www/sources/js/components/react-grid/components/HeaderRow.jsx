import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class HeaderRow extends Component {
    static propTypes = {
        onSort: PropTypes.func
    };

    static defaultProps = {
        onSort: null
    };

    constructor(props) {
        super(props);
    }

    _renderChildren() {
        const { onSort, children, sortBy, sortOrder } = this.props;

        return React.Children.map(children, child => child && React.cloneElement(child, { onSort: onSort, sortOrder: sortBy === child.props.sortBy ? sortOrder : null }));
    }

    render() {
        const { onSort, sortBy, sortOrder, ...rest } = this.props;
        return (
            <tr {...rest}>{this._renderChildren()}</tr>
        );
    }
}