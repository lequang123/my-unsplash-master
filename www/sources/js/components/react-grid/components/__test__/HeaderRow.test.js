import React from 'react';
import HeaderRow from '../HeaderRow';
import { render, shallow } from 'enzyme';

describe('HeaderRow', () => {
    it('should have correct snapshot', () => {
        // Action
        const headerRow = shallow(<HeaderRow />);

        // Assert
        expect(headerRow).toMatchSnapshot();
    });

    it('_renderChildren should return expected', () => {
        // Arrange
        const props = {
            onSort: jest.fn(),
            children: <div></div>,
            sortBy: 'somthing',
            sortOrder: 'asc'
        };
        
        const expected = React.Children.map(props.children, child => child && React.cloneElement(child, { onSort: props.onSort, sortOrder: props.sortBy === child.props.sortBy ? props.sortOrder : null }));
        const headerRow = shallow(<HeaderRow {...props} />);

        // Action
        const result = headerRow.instance()._renderChildren();

        // Assert
        expect(result).toEqual(expected);
    });

    it('render should return expected', () => {
        // Arrange
        const children = 'children';
        const expected = <tr>{children}</tr>;
        const headerRow = shallow(<HeaderRow />);
        headerRow.instance()._renderChildren = jest.fn(() => children);

        // Action
        const result = headerRow.instance().render();

        // Assert
        expect(result).toEqual(expected);
    });
});