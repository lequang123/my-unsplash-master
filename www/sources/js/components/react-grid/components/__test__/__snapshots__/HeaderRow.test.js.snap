// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderRow should have correct snapshot 1`] = `ShallowWrapper {}`;
